src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,761 @@
+   type should specify the field’s BigQuery type.
+
+2. Use the `schema` parameter to provide your table schema when you apply a
+   write transform. Set the parameter’s value to the string.
+
+
+<!-- Common -->
+
+The following example shows how to use a string to specify the same table schema
+as the previous example.
+
+```java
+{% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQuerySchemaJson
+%}```
+```py
+{% github_sample /apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_schema
+%}```
+
+
+### Setting the insertion method
+
+{:.language-py}
+> The Beam SDK for Python does not currently support specifying the insertion
+method.
+
+BigQueryIO supports two methods of inserting data into BigQuery: load jobs and
+streaming inserts. Each insertion method provides different tradeoffs of cost,
+quota, and data consistency. See the BigQuery documentation for
+[load jobs](https://cloud.google.com/bigquery/loading-data) and
+[streaming inserts](https://cloud.google.com/bigquery/streaming-data-into-bigquery)
+for more information about these tradeoffs.
+
+BigQueryIO chooses a default insertion method based on the input `PCollection`.
+
+BigQueryIO uses load jobs in the following situations:
+
+* When you apply a BigQueryIO write transform to a bounded `PCollection`.
+* When you apply a BigQueryIO write transform to an unbounded `PCollection` and
+  use `BigQueryIO.write().withTriggeringFrequency()` to set the triggering
+  frequency.
+* When you specify load jobs as the insertion method.
+
+BigQueryIO uses streaming inserts in the following situations:
+
+* When you apply a BigQueryIO write transform to an unbounded `PCollection` and
+  do not set the triggering frequency.
+* When you specify streaming inserts as the insertion method.
+
+When you use streaming inserts, you can decide what to do with failed records.
+You can either keep retrying, or return the failed records in a separate
+`PCollection`.
+
+<!-- Java specific -->
+
+{:.language-java}
+You can use `withMethod` to specify the desired insertion method. See
+[Write.Method]({{site.baseurl}}/documentation/sdks/javadoc/{{ site.release_latest }}/index.html?org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.Write.Method.html)
+for the list of the available methods and their restrictions.
+
+{:.language-java}
+***Note:*** If you use batch loads in a streaming pipeline, you must use
+`withTriggeringFrequency` to specify a triggering frequency.
+
+
+### Writing to a table
+
+{:.language-java}
+To write to a BigQuery table, apply either a `writeTableRows` or `write`
+transform.
+
+{:.language-py}
+To write to a BigQuery table, apply either a `Write` or `WriteToBigQuery` transform.
+`Write` supports batch mode. `WriteToBigQuery` supports both batch mode and
+streaming mode, but it is less efficient for batch mode. You must apply the
```

Comment:
"less efficient for batch mode" this is not correct since Python DataflowRunner replaces WriteToBigQuery with Write at runtime. I think you can only mention WriteToBigQuery here (which we will be update with a custom BQ sink hopefully in the near future). We will be adding a ReadFromBigQuery (custom BQ source) as well but we can update the documentation once we have that and when we want to make it the default.

Reply:
Here's the place where we perform transform replacement for DataflowRunner: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L483

Reply:
Oh, okay, so bullet 2 in https://issues.apache.org/jira/browse/BEAM-3289 is done? in that case I will remove it and have WriteToBigQuery as the only option.

Reply:
Yeah, it was already done. Added  a comment to the JIRA. Sorry about the confusion.